### PR TITLE
fix: slow git status on wsl2

### PR DESCRIPTION
**Pull request by evilz**
Should fix this issue #505

use windows git.exe for mounted drive

### CODE REVIEW

File: src/segment_git.go [hunk 1]
```diff
@@ -272,7 +272,7 @@ func (g *git) getStatusColor(defaultValue string) string {
 
 func (g *git) getGitCommandOutput(args ...string) string {
 	gitCommand := "git"
-	if g.env.getRuntimeGOOS() == windowsPlatform {
+	if g.env.getRuntimeGOOS() == windowsPlatform || (g.env.isWsl() && strings.HasPrefix(g.env.getcwd(), "/mnt/")) {
```
Comment (JanDeDobbeleer) [inline src/segment_git.go]: Does it matter where it runs in this case?

**Comment by evilz** [comment]
Yes, it's all about fileshare.
Default linux git command should be use outside Windows mount.

Le mer. 10 mars 2021 à 19:05, Jan De Dobbeleer <notifications@github.com> a
écrit :

File: src/segment_git.go (hunk 1 above)
Comment (JanDeDobbeleer) [reply src/segment_git.go]: ok, can we in that case add an inline function for clarity?

```golang
func (g *git) getGitCommandOutput(args ...string) string {
        inWSLSharedDrive := func(env environmentInfo) bool {
              return env.isWsl() && strings.HasPrefix(env.getcwd(), "/mnt/")
        }
	gitCommand := "git"
	if g.env.getRuntimeGOOS() == windowsPlatform || inWSLSharedDrive(g.env) {
		gitCommand = "git.exe"
	}
	args = append([]string{"--no-optional-locks", "-c", "core.quotepath=false", "-c", "color.status=false"}, args...)
	val, _ := g.env.runCommand(gitCommand, args...)
	return val
}
```

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Comment by JanDeDobbeleer** [comment]
@evilz can you make 1 commit out of this and format the file correctly with gofmt? Thx!

**Comment by evilz** [comment]
I think `environmentInfo` should be more typed.

Like `getcwd` should return something else that a string where we can call inWSLSharedDrive on it.

Also `getRuntimeGOOS()` and `getPlatform()` can return enum or something more typed. 

That only my my two cents :P

**Pull request merged**